plugins/storage/volume/solidfire/src/org/apache/cloudstack/storage/datastore/provider/SolidFireHostListener.java

```diff
@@ -65,22 +94,182 @@ public boolean hostConnect(long hostId, long storagePoolId) {
             storagePoolHostDao.persist(storagePoolHost);
         }
 
-        // just want to send the ModifyStoragePoolCommand for KVM
-        if (host.getHypervisorType() != HypervisorType.KVM) {
-            return true;
+        if (host.getHypervisorType().equals(HypervisorType.XenServer)) {
+            handleXenServer(host.getClusterId(), host.getId(), storagePoolId);
+        }
+        else if (host.getHypervisorType().equals(HypervisorType.KVM)) {
+            handleKVM(hostId, storagePoolId);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean hostDisconnected(long hostId, long storagePoolId) {
+        StoragePoolHostVO storagePoolHost = storagePoolHostDao.findByPoolHost(storagePoolId, hostId);
+
+        if (storagePoolHost != null) {
+            storagePoolHostDao.deleteStoragePoolHostDetails(hostId, storagePoolId);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean hostAboutToBeRemoved(long hostId) {
+        HostVO host = _hostDao.findById(hostId);
+
+        handleVMware(host, false);
```

Comment:
Unfortunately, it appears to be too late at that point because CloudStack can't talk to the host anymore.